packages/vite/src/node/utils.ts

```diff
@@ -616,3 +616,58 @@ export const usingDynamicImport = typeof jest === 'undefined'
 export const dynamicImport = usingDynamicImport
   ? new Function('file', 'return import(file)')
   : require
+
+/**
+ * Indicates whether the value is surrounded by quotes.
+ *
+ * ---
+ *
+ * Example with double quotes:
+ * ```
+ * a(href="#")
+ * ```
+ *
+ * In this case `val` is `"#"`.
+ *
+ * ---
+ *
+ * Example with single quotes:
+ * ```
+ * a(href='#')
+ * ```
+ *
+ * In this case `val` is `'#'`.
+ *
+ * ---
+ *
+ * Example with no quotes:
+ * ```
+ * - const route = '#';
+ * a(href=route)
+ * ```
+ *
+ * In this case `val` is `route`.
+ *
+ * ---
+ *
+ * Special cases:
+ * ```
+ * a(href='/' + '#')
+ * a(href="/" + "#")
+ * ```
+ *
+ * These cases should not be treated as quoted.
+ *
+ * ---
+ *
+ * @param val Value of tag attribute.
+ * @returns Whether the value is quoted or not.
+ */
```

Comment:
Ahaha, please remove my JSDoc from `@prettier/plugin-pug` 😅 
It's completely unrelated to Vite

```suggestion
```

Reply:
Ok -。-